### PR TITLE
[ISSUE-510] Fix componentWillReceiveProps deprecation warning

### PR DESCRIPTION
**Pull request by jonathanconway**

### CODE REVIEW

**Comment by giamir** [comment]
Any plan to use the new `getDerivedStateFromProps` API instead of prefixing the deprecated lifecycle hook with `UNSAFE`?

File: src/Cleave.react.js [hunk 1]
```diff
@@ -23,7 +23,7 @@ var cleaveReactClass = CreateReactClass({
         Util.setSelection(owner.element, owner.state.cursorPosition, pps.document);
     },
 
-    componentWillReceiveProps: function (nextProps) {
+    UNSAFE_componentWillReceiveProps: function (nextProps) {
```
Comment (eduardoflorendo) [inline src/Cleave.react.js]: #

**Comment by cht8687** [comment]
@giamir  

Seems this https://github.com/nosir/cleave.js/pull/527 fixed the issue so that we no longer need this fix.  thoughts?

**Comment by giamir** [comment]
@cht8687 True; #527 is still open at the moment.

File: src/Cleave.react.js (hunk 1 above)
Comment (eduardoflorendo) [reply src/Cleave.react.js]: Doing interaction i've experience a props make you mislead and then when commit an error on the recent pages your information you created will disaffear in that case your personal information including credit card numbers names remain in the files hackers easily download your document and using for fraud..

**Comment by cht8687** [comment]
thanks @jonathanconway 

hey @giamir , we've merged #527 and I think we can close this now. We will release a new version soon.

**Comment by giamir** [comment]
Awesome @cht8687 . Thanks for your help.

**Pull request closed without merging**